distributed/tests/test_core.py

```diff
@@ -1380,3 +1381,78 @@ async def test_async_listener_stop(monkeypatch):
         async with Server({}) as s:
             await s.listen(0)
             assert s.listeners
+
+
+@gen_test()
+async def test_messages_are_ordered_bsend():
+    ledger = []
+
+    async def async_handler(val):
+        await asyncio.sleep(0.01)
```

Comment:
I'd randomize/adjust the sleep here to test that async handlers remain in order not just because they were started in order.